sympy/assumptions/handlers/sets.py

```diff
@@ -770,3 +770,26 @@ def _(expr, assumptions):
     x = expr.args[0]
     if ask(Q.algebraic(x), assumptions):
         return ask(~Q.nonzero(x - 1), assumptions)
+
+
+# TranscendentalPredicate
+
+@TranscendentalPredicate.register_many(Exp1, Pi, GoldenRatio)
+def _(expr, assumptions):
+    return True
+
+@TranscendentalPredicate.register_many(AlgebraicNumber, Float, TribonacciConstant,
```

Comment:
We should omit Float here I think. We intentionally leave Floats as None for rational/irrational.

Reply:
We should probably omit Float in the handlers for algebraic as well. Currently this is not omitted:
```
>>> f = sympify(1.5)
>>> ask(Q.algebraic(f))
True
```
yet
```
>>>  print(f.is_algebraic)
None
```

Reply:
@asmeurer what do think about this?

Reply:
Removing `float` from here results in Q.transcendental(1.0) as None.